### PR TITLE
new compset naming convention

### LINKED ISSUE #4125: Need for a new compset naming convention
I would like to add a new component to cesm and thus to cime for the cmeps driver.   This component would be optional and only initially needed for a few compsets.  I do not want to repeat the experience of adding the IAC and ESP components which have created a bit of spaghetti code in this repository and in cmeps.    See also https://github.com/ESCOMP/CMEPS/issues/254

### PR DESCRIPTION
**Pull request by jedwards4b**
Introduce a new compset naming convention which maintains backward compatibility and allows position independence.
This convention uses a three character lower case prefix to indicate the component class (tim, atm, lnd, ice, ...) followed by a : .  Some examples are:  

- tim:2000_atm:DATM%NYF_ocn:DOCN%SOMAQP
- tim:1850_lnd:DLND%SCPL
- atm:DATM%NYF_ice:DICE%SSMI_ocn:DOCN%DOM_rof:DROF%NYF_tim%2000

Test suite:
Test baseline:
Test namelist changes:
Test status: [bit for bit, roundoff, climate changing]

Fixes #4125 

User interface changes?:

Update gh-pages html (Y/N)?:

### CODE REVIEW

**Comment by rljacob** [comment]
@golaz, @jonbob, @bishtgautam, @wlin7 please be aware of this.  Too late for v2 but could be introduced in v3.

**Comment by ekluzek** [comment]
@jedwards4b thanks for putting this together and the discussion on #4125. This will solve a bunch of problems and make it so much easier to parse these longnames. So I'm really looking forward to taking advantage of it.

The one suggestion I have is to maybe change the name of "tim" (short for time) to something else as it also covers SSP-RCP scenarios, and could include Paleo scenarios and things like Aquaplanet and Held-Suarez. So maybe scn for (scenario)?

**Comment by mvertens** [comment]
@ekluzek - very good point. I agree. How about scn% for scenario?

**Comment by jedwards4b** [comment]
Okay I'll change tim to scn but it's : not %.

**Comment by mvertens** [comment]
Right - scn:

**Comment by jedwards4b** [comment]
@fischer-ncar could you run prealpha tests against this branch please?

**Review by jgfouca — APPROVED** [review]
Looks good

**Comment by fischer-ncar** [comment]
@jedwards4b Okay, I'm setting up prealpha tests on cheyenne.

File: scripts/lib/CIME/XML/grids.py [hunk 1]
```diff
@@ -57,6 +57,8 @@ def get_grid_info(self, name, compset, driver):
         atmnlev = None
         lndnlev = None
 
+        print("compset is now {} ".format(compset))
```
Comment (billsacks) [inline scripts/lib/CIME/XML/grids.py:60]: Just checking: did you mean to leave this in here, or should this be changed to something like a logging.debug?

**Review by billsacks — COMMENTED** [review]
I just did a cursory review of this. One comment below. Thanks a lot for your work on this - this will be really nice to have moving forward!

**Comment by fischer-ncar** [comment]
@jedwards4b The prealpha tests on cheyenne passed.

**Review by fischer-ncar — APPROVED** [review]

**Pull request merged**